### ISSUE TITLE
Update the migration doc with the latest changes for v3+ plugin 

### ISSUE THREAD

**Issue by camilamacedo86** [kubernetes-sigs/kubebuilder#1829]
**What do you want to happen?**

Before we make the plugin v3+ stable we need update the doc : https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/migration/migration_guide_v2tov3.md#migrating-your-projects-to-use-v3-plugins

Is there currently another issue associated with this? 

See that we have some changes which require to be addressed: 
-  https://github.com/kubernetes-sigs/kubebuilder/pull/1644 - new PROJECT file attributes: crdVersion and webhookVersion 

/kind feature

**Comment by camilamacedo86** [comment]
Blocked by https://github.com/kubernetes-sigs/kubebuilder/pull/1899
c/c @estroz
